routes/serve-routes.js

```diff
@@ -1,39 +1,47 @@
 const logger = require('winston');
-const { getAssetByClaim, getChannelContents, getAssetByChannel, serveOrShowAsset } = require('../controllers/serveController.js');
+const { getClaimId, getChannelContents, getLocalFileRecord, getClaimRecord } = require('../controllers/serveController.js');
+const serveHelpers = require('../helpers/serveHelpers.js');
 const { handleRequestError } = require('../helpers/errorHandlers.js');
+const db = require('../models');
 
 const SERVE = 'SERVE';
 const SHOW = 'SHOW';
 const SHOWLITE = 'SHOWLITE';
-const CHANNEL = 'CHANNEL';
-const CLAIM = 'CLAIM';
 const CLAIM_ID_CHAR = ':';
 const CHANNEL_CHAR = '@';
 const CLAIMS_PER_PAGE = 10;
 const NO_CHANNEL = 'NO_CHANNEL';
 const NO_CLAIM = 'NO_CLAIM';
+const NO_FILE = 'NO_FILE';
 
 function isValidClaimId (claimId) {
   return ((claimId.length === 40) && !/[^A-Za-z0-9]/g.test(claimId));
 }
 
 function isValidShortId (claimId) {
-  return claimId.length === 1;  // really it should evaluate the short url itself
+  return claimId.length === 1;  // it should really evaluate the short url itself
 }
 
 function isValidShortIdOrClaimId (input) {
   return (isValidClaimId(input) || isValidShortId(input));
 }
 
-function getAsset (claimType, channelName, channelId, name, claimId) {
-  switch (claimType) {
-    case CHANNEL:
-      return getAssetByChannel(channelName, channelId, name);
-    case CLAIM:
-      return getAssetByClaim(name, claimId);
-    default:
-      return new Error('that claim type was not found');
+function isUriAChannel (name) {
+  return (name.charAt(0) === CHANNEL_CHAR);
+}
+
+function returnChannelNameFromUri (uri) {
```

Comment:
You may want to look at using `lbryuri.js` for some of this: https://github.com/lbryio/lbry-app/blob/master/src/renderer/js/lbryuri.js

In particular, the `parse` and `build` functions (which I recommend just trying to call so you see what they do if reading the code is tricky).

Reply:
this was very helpful.  I made my own version since the uri on spee.ch is slightly different, but used this pattern.  Probably could still be cleaned up a bit, but it simplified the code a lot.